twigs/ecr.py

```diff
@@ -0,0 +1,151 @@
+
+                i_json_images = run_aws_cmd(iImage_cmd)
+                for image in i_json_images['imageDetails']:
+
+                    if args.tag is None:
+                        no_of_tags = len(image['imageTags'])
+                        if 'latest' in image['imageTags']:
+                            tag = ':' + 'latest'
+                        else:
+                            tag = ':' + image['imageTags'][no_of_tags -1]
+
+
+                    logging.info("Using %s as a latest tag for the repository %s", tag,repo['repositoryName'])            
+                    args.image = get_repo_uri(repo['repositoryName'],args.repositoryType) + tag
+                    args.assetid = args.image
+                    args.assetname = args.image
+                    logging.info("Discovering image %s", args.image)
+                    assets = docker.get_inventory(args)
+
+                    if assets:
+                        allassets.extend(assets)            
+        for a in allassets:
+            a['tags'].append('ECR')
+            return allassets
+    else:
+        #particular image(s) in the repository        
+        tokens = args.repositoryUri.split('.amazonaws.com/')
+        uri = tokens[0]
+        repo_name = tokens[1]
+
+        if ':' not in repo_name:#get the latest tag for image, if available 'latest' otherwise most recent.
+        
+            if args.repositoryType == 'private':
+                ilist_cmd = "ecr describe-images --repository-name " + repo_name 
+            else:
+                ilist_cmd = "ecr-public describe-images --repository-name " + repo_name
+       
+            i_json = run_aws_cmd(ilist_cmd)
+
+            if len(i_json) > 0:
+
+                for image in i_json['imageDetails']:
+                    if args.tag is None:
+                        no_of_tags = len(image['imageTags'])
+                        if 'latest' in image['imageTags']:
+                            tag = ':' + 'latest'
+                        else:
+                            tag = ':' + image['imageTags'][no_of_tags-1]
+                
+                    logging.info("Using %s as a latest tag for the repository %s", tag,repo_name)            
+                    args.image = get_repo_uri(repo_name,args.repositoryType) + tag
+                    args.assetid = args.image
+                    args.assetname = args.image
+                    logging.info("Discovering image %s", args.image)
+                    assets = docker.get_inventory(args)
+
+                    if assets:
+                        allassets.extend(assets)            
+            for a in allassets:
+                a['tags'].append('ECR')
+                return allassets
+            else:
+                logging.info("This repository has no images")
+                return None
+        else:#user has provided repositoryUri (image) with tag
```

Comment:
I am wondering whether we need separate "tag" argument on twigs command-line, if one can specify tag along with registryUri. Let us discuss.

Reply:
no, we don't need a 'tag' argument anymore.